include/clap/ext/draft/location.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "../../color.h"
+#include "../../plugin.h"
+#include "../../string-sizes.h"
+
+// This extension allows a host to tell the plugin more about its position
+// within a project or session.
+
+static CLAP_CONSTEXPR const char CLAP_EXT_LOCATION[] = "clap.location/1";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   // Represents a document/project/session.
+   CLAP_PLUGIN_LOCATION_PROJECT = 1,
+
+   // Represents a group of tracks.
+   // It can contain both track groups, tracks and devices (post processing).
+   // The first device within a track group has the index of
+   // the last track or track group within this group + 1.
+   CLAP_PLUGIN_LOCATION_TRACK_GROUP = 2,
+
+   // Represents a single tracks. It contains devices (serial).
+   CLAP_PLUGIN_LOCATION_TRACK = 3,
+
+   // Represents a single device.
+   // It contains other nested device chains.
+   CLAP_PLUGIN_LOCATION_DEVICE = 4,
+
+   // Represents a nested device chain (serial).
+   // Its parent must be a device.
+   // It contains other devices.
+   CLAP_PLUGIN_LOCATION_NESTED_DEVICE_CHAIN = 5,
```

Comment:
Is there a reason why nested device chains & devices function differently from how track groups & tracks function? That is, why do device chains need to be a child of a device rather than function as a "device group" which can have child devices?

Reply:
A device chain is a serial list of devices.
A device can contain multiple device chains, which maybe use for parallel processing.
There are similarities between tracks and device chain. Though there's a different semantic, which the plugin can use for display. Think Fabfilter Pro Q3, with sidechain.